### PR TITLE
CI Linux: Consolidate jobs

### PR DESCRIPTION
**Pull request by mkoeppe**
Thanks to the more powerful runners in use since early 2024, we can do away with the splits between
- `standard-pre` and `standard`
- `minimal-pre` and `minimal`
- `optional-0-o` and `optional-p-z`
- `experimental-0-o` and `experimental-p-z`

This makes the CI runs a bit easier to navigate for those who look at them.

- Test run: https://github.com/mkoeppe/sage/actions/runs/10438362191

This should also fix "No space left on device" aborts as seen for example in https://github.com/mkoeppe/sage/actions/runs/10426436105 for standard-pre (where we did not use to maximize the available disk space).

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

- Depends on #37434 (merged here)

### CODE REVIEW

File: .github/workflows/ci-linux.yml [hunk 1]
```diff
@@ -54,45 +54,27 @@ jobs:
...
-      # FIXME: duplicated from env.TARGETS
-      targets_pre: all-sage-local
-      tox_packages_factors: >-
-          ["standard"]
-      docker_push_repository: ghcr.io/${{ github.repository }}/
-      # Make sure that all "standard-pre" jobs can start simultaneously,
-      # so that runners are available by the time that "default" starts.
-      max_parallel: 50
-
   standard:
```
Comment (kwankyu) [inline .github/workflows/ci-linux.yml:57]: Fix L37.

**Comment by kwankyu** [comment]
This simplification is very welcome. I am just curious what was the purpose of `-pre` jobs and why "more powerful runners" allows us to dispense with them.

**Comment by mkoeppe** [comment]
Before the runners were upgraded, the jobs did not fit in the 6 hour time limit.

**Comment by kwankyu** [comment]
Ah, I see. 

No merge conflict with #37434?

**Comment by mkoeppe** [comment]
> No merge conflict with #37434?

I've merged it now to make the adjustment c856eed

**Comment by kwankyu** [comment]
This looks good to me.

Let me know when the test run of the (merged) workflow is done.

**Review by kwankyu — APPROVED** [review]

File: .github/workflows/ci-linux.yml [hunk 2]
```diff
@@ -189,41 +171,11 @@ jobs:
...
-    with:
-      incremental: true
-      free_disk_space: true
-      from_docker_repository: ghcr.io/${{ github.repository }}/
-      from_docker_target: "with-targets-pre"
-      tox_packages_factors: >-
-          ["maximal"]
-      docker_targets: "with-targets-optional"
-      targets_optional: '$(echo $(export PATH=build/bin:$PATH && sage-package list :experimental: --has-file "spkg-install.in|spkg-install|requirements.txt" --no-file "huge|has_nonfree_dependencies" | grep -v sagemath_doc | grep ^[0-9a-o]))'
+      targets_optional: '$(echo $(export PATH=build/bin:$PATH && sage-package list :optional: --has-file "spkg-install.in|spkg-install|requirements.txt" --no-file "huge|has_nonfree_dependencies" | grep -v sagemath_doc))'
```
Comment (kwankyu) [inline .github/workflows/ci-linux.yml]: may remove L173.

**Comment by dcoudert** [comment]
@fchapoton fixed the lint errors in #38518

File: .github/workflows/ci-linux.yml (hunk 2 above)
Comment (mkoeppe) [reply .github/workflows/ci-linux.yml]: Thanks - actually I had to fix something that that comment explained

Comment (kwankyu) [reply .github/workflows/ci-linux.yml]: Ah really.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**